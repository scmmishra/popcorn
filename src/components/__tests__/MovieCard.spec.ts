import { describe, it, expect, beforeEach } from "vitest";

import { mount, VueWrapper } from "@vue/test-utils";
import MovieCardVue from "../MovieCard.vue";

const mockMovie = () => {
  return {
    adult: false,
    backdrop_path: "/yNXarQfosBZADXVzdLlvDjsk8aq.jpg",
    genre_ids: [18, 35],
    id: 914216,
    original_language: "en",
    original_title: "The Lost King",
    overview:
      "An amateur historian defies the academic establishment in her efforts to find King Richard III's remains, which were lost for over 500 years.",
    popularity: 6.896,
    poster_path: "/bvteSajbZguotujz0e4Y8JpgJ1G.jpg",
    release_date: "2022-10-07",
    title: "The Lost King",
    video: false,
    vote_average: "8",
    vote_count: 5,
  };
};

describe("MovieCard", () => {
  let wrapper: VueWrapper;

  beforeEach(() => {
    wrapper = mount(MovieCardVue, {
      props: mockMovie(),
      global: {
        stubs: {
          MovieRating: true,
        },
      },
    });
  });

  it("matches snapshot", () => {
    expect(wrapper.html()).toMatchSnapshot();
  });

  it("renders properly", () => {
    expect(wrapper.text()).toContain("The Lost King");
  });
});
