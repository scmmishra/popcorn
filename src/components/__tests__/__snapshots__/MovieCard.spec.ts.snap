// Vitest Snapshot v1

exports[`MovieCard > matches snapshot 1`] = `
"<div class=\\"movie-card\\" adult=\\"false\\" backdrop_path=\\"/yNXarQfosBZADXVzdLlvDjsk8aq.jpg\\" genre_ids=\\"18,35\\" id=\\"914216\\" original_language=\\"en\\" original_title=\\"The Lost King\\" overview=\\"An amateur historian defies the academic establishment in her efforts to find King Richard III's remains, which were lost for over 500 years.\\" popularity=\\"6.896\\" video=\\"false\\" vote_count=\\"5\\" data-v-1fcbeb8b=\\"\\"><img src=\\"https://image.tmdb.org/t/p/w500//bvteSajbZguotujz0e4Y8JpgJ1G.jpg\\" alt=\\"The Lost King\\" data-v-1fcbeb8b=\\"\\">
  <div class=\\"movie-card-overlay\\" data-v-1fcbeb8b=\\"\\">
    <div class=\\"rating-container\\" data-v-1fcbeb8b=\\"\\">
      <movie-rating-stub value=\\"8\\" data-v-1fcbeb8b=\\"\\"></movie-rating-stub>
    </div>
    <div class=\\"info-container\\" data-v-1fcbeb8b=\\"\\">
      <h4 data-v-1fcbeb8b=\\"\\">The Lost King</h4>
      <p data-v-1fcbeb8b=\\"\\">October 2022</p>
    </div>
  </div>
  <div class=\\"mobile-info\\" data-v-1fcbeb8b=\\"\\">
    <!-- this replication removes a lot of uncessary CSS -->
    <h4 data-v-1fcbeb8b=\\"\\">The Lost King</h4>
    <p data-v-1fcbeb8b=\\"\\">October 2022</p>
  </div>
</div>"
`;
